### ISSUE TITLE
[BUG] Query returns null values when dataframe is not constructed from pandas

### ISSUE THREAD

**Issue by beckernick** [rapidsai/cudf#1994]
If we create the gpu dataframe using `from_pandas`, it works as expected, but if we create the gpu dataframe directly, we return nulls in some cases.

It appears to be sensitive to interaction of the location of the nulls and the conditional expression:

```python
import cudf
import pandas as pd
​
pdf = pd.DataFrame({'a':[0,1.0,2.,None,None,3,5]})
gdf = cudf.from_pandas(pdf)
print(gdf.query("a < 3"))
     a
0  0.0
1  1.0
2  2.0
```
```python
import cudf
import pandas as pd
​
gdf = cudf.DataFrame({'a':[0,1.0,2.,None,None,3,5]})
print(gdf.query("a < 3"))
     a
0  0.0
1  1.0
2  2.0
3     
4     
```

```python
import cudf
import pandas as pd
​
gdf = cudf.DataFrame({'a':[0,1.0,2.,None,3,None,4]})
print(gdf.query("a != 3"))
     a
0  0.0
1  1.0
2  2.0
3     
5     
6  4.0
```

```python
import cudf
import pandas as pd
​
gdf = cudf.DataFrame({'a':[0,1.0,2.,None,3,None,4, None, 9]})
print(gdf.query("a > 3"))
     a
6  4.0
8  9.0
```

Caveat: I'm in the nightly container from dockerhub, which is two days old

**Comment by beckernick** [comment]
Update: This is expected behavior, but we should have a guard against this.

**Comment by kkraus14** [comment]
@cwharris This is likely related to a combination of using `to_gpu_array()` instead of `.data.mem` as well as not carrying over the existing nullmask of queried columns and taking that into account.

**Comment by cwharris** [comment]
The results of these query expressions seems off to me. Why is `None` less than anything? Shouldn’t any operation involving ‘None` result in either `None` or `false`? Example: `patrons = person form people if people.age >= 21` would  include people under 21 if we hadn’t determined an age for them yet. That’s is undesirable. Do we have a spec for this, or are we targeting whatever PANDAS would do?

Should `from_pandas` be filtering out those rows with null in the column?

Some of the reported behavior is not what I would expect.

**Comment by cwharris** [comment]
Binary comparisons are bugged with regards to nulls. https://github.com/rapidsai/cudf/issues/2366

**Comment by cwharris** [comment]
#2366 fixes this for numerical columns, but not `query`, specifically. I'm looking in to this now.

**Comment by cwharris** [comment]
This bug can also be reproduced with `cudf.from_pandas(pdf, nan_as_null=True)`.
```
 # succeed pass if `flag` is `False` / `None`
pdf = pd.DataFrame(data)
gdf = cudf.from_pandas(pdf, nan_as_null=flag)
assert_eq(pdf, gdf)
```

Since the `Dataframe` constructor sees the original data with nulls, it treats them as `None`.
```
 # will fail if `data` has a null value in any column.
pdf = pd.DataFrame(data)
gdf = cudf.DataFrame(data)
assert_eq(pdf, gdf)
```

**Comment by cwharris** [comment]
@beckernick @kkraus14 

I have a fix for this which simply invalidates the results which had any invalid inputs, but I'm uneasy about it because it doesn't handle `!=` or mulit-condition operations correctly, nor is it possible to handle them correctly for all dtypes without comparing the null bits on and individual operator level.

Another option is traversing the query's ast (which we already have in queryutils) and creating a complementary validity query to run along side the original potentially in the same kernel.

Thoughts?

#### Fails:
```
def test_multi_condition():
    data = {
        'a':[0., -100., 0., 100.],
        'b':[0., None, None, None]
    }

    pdf = pd.DataFrame(data)
    gdf = cudf.DataFrame(data)

    query = "a >= 0 or b <= 0"
    pdf_q_res = pdf.query(query)
    gdf_q_res = gdf.query(query)

    print(pdf)
    print(gdf)
    print()
    print(pdf_q_res)
    print(gdf_q_res)
    print()

    assert_eq(pdf_q_res, gdf_q_res)
```
```
       a    b
0    0.0  0.0
1 -100.0  NaN
2    0.0  NaN
3  100.0  NaN
        a    b
0     0.0  0.0
1  -100.0     
2     0.0     
3   100.0     

       a    b
0    0.0  0.0
2    0.0  NaN
3  100.0  NaN
     a    b
0  0.0  0.0


```

**Comment by cwharris** [comment]
Fixed in #2466
